model/generate.go

```diff
@@ -184,6 +185,22 @@ func updateParserProject(ctx context.Context, settings *evergreen.Settings, v *V
 		return nil
 	}
 
+	// If this is the first time the parser project has been updated by a generator, cache a copy of
+	// the parser project representing its state before any generators updated it.
```

Comment:
Might be more helpful to state why we need to do this (i.e. special case to hold onto pre-generated copy for child patches that may call generate.tasks again) rather than just what this logic is doing.